sklearn/metrics/tests/test_pairwise.py

```diff
@@ -370,6 +373,68 @@ def test_pairwise_distances_argmin_min():
     np.testing.assert_almost_equal(dist_orig_val, dist_chunked_val, decimal=7)
 
 
+def test_pairwise_distances_reduce_invalid_reduce_func():
+    X = np.empty((400, 4))
+    y = np.empty((200, 4))
+    assert_raise_message(ValueError, 'reduce_func needs to be passed as an '
+                         'argument', pairwise_distances_reduce, X, y,
+                         block_size=0, metric='euclidean')
+
+
+def _reduce_func(dist):
+    return dist[:, :100]
+
+
+def test_pairwise_distances_reduce():
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((400, 4))
+    # Reduced Euclidean distance
+    S = pairwise_distances(X)[:, :100]
+    S2 = pairwise_distances_reduce(X, None, reduce_func=_reduce_func,
+                                   block_size=1)
+    assert_array_almost_equal(S, S2)
+
+
+def test_pairwise_distances_blockwise_invalid_block_size():
+    X = np.empty((400, 4))
+    y = np.empty((200, 4))
+    assert_raise_message(ValueError, 'block_size should be at least n_samples '
+                         '* 8 bytes = 1 MiB, got 0',
+                         pairwise_distances_blockwise, X, y, block_size=0,
+                         metric='euclidean')
+
+
+def check_pairwise_distances_blockwise(X, Y, block_size, metric):
+    gen = pairwise_distances_blockwise(X, Y, block_size=block_size,
+                                       metric=metric)
+    blockwise_distances = np.vstack(list(gen))
```

Comment:
You also need to test that each component does not take more than the specified memory.